### PR TITLE
[BE] Fix signed-unsigned warnings

### PR DESCRIPTION
**Pull request by malfet**
Switch to range loops when possible
Replace `ptrdiff_t`(signed type) with `size_t`(unsigned type)

### CODE REVIEW

File: aten/src/TH/generic/THStorageCopy.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@ void THStorage_(copy)(THStorage *storage, THStorage *src)
   scalar_t *scalar_src = THStorage_(data)(src);
   scalar_t *data = THStorage_(data)(storage);
   uint64_t numel = storage->nbytes() / sizeof(scalar_t);
-  for (ptrdiff_t i = 0; i < numel; ++i) {
+  for (uint64_t i = 0; i < numel; ++i) {
```
Comment (walterddr) [inline aten/src/TH/generic/THStorageCopy.cpp:11]: nit: can `ptrdiff_t` size change based on platform?

**Review by walterddr — APPROVED** [review]

File: aten/src/TH/generic/THStorageCopy.cpp (hunk 1 above)
Comment (malfet) [reply aten/src/TH/generic/THStorageCopy.cpp:11]: Yes, it is 32-bit on 32-bit OSes, but I don't think PyTorch supports any at the moment.

**Comment by mruberry** [comment]
Unlanding. This broke several builds. Example snippet:

```
test_conv_transposed_large_cuda - TestNNDeviceTypeCUDA
test_nn.py

Traceback (most recent call last):
  File "C:\Users\circleci\project\build\win_tmp\build\torch\testing\_internal\common_device_type.py", line 273, in instantiated_test
    result = test_fn(self, *args)
  File "C:\Users\circleci\project\build\win_tmp\build\torch\testing\_internal\common_device_type.py", line 651, in only_fn
    return fn(slf, device, *args, **kwargs)
  File "C:\Users\circleci\project\build\win_tmp\build\torch\testing\_internal\common_device_type.py", line 612, in dep_fn
    return fn(self, *args, **kwargs)
  File "test_nn.py", line 11312, in test_conv_transposed_large
    ret = conv(input_large)
  File "C:\Users\circleci\project\build\win_tmp\build\torch\nn\modules\module.py", line 744, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\circleci\project\build\win_tmp\build\torch\nn\modules\conv.py", line 832, in forward
    output_padding, self.groups, self.dilation)
RuntimeError: CUDA out of memory. Tried to allocate 4.00 GiB (GPU 0; 14.85 GiB total capacity; 12.00 GiB already allocated; 1.65 GiB free; 12.00 GiB reserved in total by PyTorch)
```

This could be due to changes like this loop:

```
for (int64_t i = 1; i < o.size(); i++) {	 
  outsize *= o[i];
}
```

becoming this one:

```
for (const auto& e: o) {
  outsize *= e;
}
```

The previous loop skipped the first element in `o`.

**Comment by mruberry** [comment]
Style nit: use decltype when declaring loop variables.

File: aten/src/ATen/native/Convolution.cpp [hunk 2]
```diff
@@ -177,13 +177,13 @@ auto ConvParams::needs_64bit_indexing_no_split(const at::Tensor& input, const at
   int64_t outsize = 1;
   if (transposed) {
     std::vector<int64_t> o = conv_input_size(input.sizes(), weight.sizes(), padding, output_padding, stride, dilation, groups);
-    for (int64_t i = 1; i < o.size(); i++) {
-      outsize *= o[i];
+    for (const auto& e: o) {
```
Comment (ngimel) [inline aten/src/ATen/native/Convolution.cpp:180]:
```suggestion
    int64_t outsize = prod_intlist(o.begin()+1, o.end())
```
similar in the next loop

Comment (malfet) [reply aten/src/ATen/native/Convolution.cpp]: Applied!

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**